### DISCUSSION TITLE
is process.shell officially supported?

### DISCUSSION THREAD

**Discussion by briankegerreis** [nextflow-io/nextflow#2337] [Q&A]
I am trying to set a pipeline to run all script blocks with pipefail, and a solution I have seen is to set process.shell to ["/bin/bash", "-ueo", "pipefail"]. However, process.shell isn't documented, and I have only seen it pop up in some nf-core pipelines and on Gitter.  Is this safe to use, and if so, is there a reason it is not in the documentation?

https://github.com/nextflow-io/nextflow/blob/7e7d52573379ffdd831c21b5e9c3b4dd669ac800/modules/nextflow/src/main/groovy/nextflow/script/ProcessConfig.groovy#L75

**Comment by pditommaso** [comment]
You are right, it was never documented but it's safe to use.
